bson/_cbsonmodule.c

```diff
@@ -2901,11 +2901,31 @@ static PyObject* _cbson_array_of_documents_to_buffer(PyObject* self, PyObject* a
                             "not enough data for a BSON document");
             Py_DECREF(InvalidBSON);
         }
-        goto done;
+        goto fail;
     }
 
     memcpy(&size, string, 4);
     size = BSON_UINT32_FROM_LE(size);
+
+    /* validate the size of the array */
+    if (view.len < (int32_t)size || (int32_t)size < BSON_MIN_SIZE) {
```

Comment:
Should `view.len < (int32_t)size` be `view.len != (int32_t)size`?

Reply:
Fixed